pkg/controller/controller_ref_manager_test.go

```diff
@@ -178,7 +178,7 @@ func TestClaimVirtualMachineInstance(t *testing.T) {
 		claimed, err := test.manager.ClaimVirtualMachineInstances(test.virtualmachines)
 		if test.expectError && err == nil {
 			t.Errorf("Test case `%s`, expected error but got nil", test.name)
-		} else if !reflect.DeepEqual(test.claimed, claimed) {
+		} else if !equality.Semantic.DeepEqual(test.claimed, claimed) {
```

Comment:
we could make use of gofuzz to protect us better. LIke the `newVirtualMachine` function could create VMIs like this:

```go

// 50% chance of nil and 50% chance of non-nil but empty maps and arrays
vm := &VirtualMachine{}
fuzz.New().NilChance(0.5).NumElements(0,0).Fuzz(vm)
// setup test specific basics
vm.Spec.X= y
```

Reply:
Randomizing input in tests is a special class of testing. I think it will be good to have these tested in a separated job, focusing on randomized input as you just suggested.

Reply:
is this something we can follow up on later after this PR? I'd like to at least the the virt-controller's vmi.go controller updated to use semantic equals asap.

Reply:
yep, let's have an issue.